### PR TITLE
PHP tests unification

### PR DESCRIPTION
**Pull request by pdostal**
- Related ticket: [poo#109810](https://progress.opensuse.org/issues/109810)
- SLE job groups: [gitlab.suse.de/qa-maintenance/qam-openqa-yml!388](https://gitlab.suse.de/qa-maintenance/qam-openqa-yml/-/merge_requests/388)
- openSUSE job groups: os-autoinst/opensuse-jobgroups#208
- Verification run: [SLE 12-SP2](https://pdostal-server.suse.cz/tests/834), [SLE 12-SP3](https://pdostal-server.suse.cz/tests/837), [SLE 12-SP4](https://pdostal-server.suse.cz/tests/839), [SLE 12-SP5](https://pdostal-server.suse.cz/tests/835), [SLE 15](https://pdostal-server.suse.cz/tests/833), [SLE 15-SP1](https://pdostal-server.suse.cz/tests/832), [SLE 15-SP2](https://pdostal-server.suse.cz/tests/840), [SLE 15-SP3](https://pdostal-server.suse.cz/tests/838), [SLE 15-SP4](https://pdostal-server.suse.cz/tests/836), [Tumbleweed](https://pdostal-server.suse.cz/tests/831).

### CODE REVIEW

**Comment by pablo-herranz** [comment]
Well done! Thanks for finishing my job :)

File: tests/console/flask.pm [hunk 1]
```diff
@@ -44,11 +44,14 @@ sub run {
...
-    assert_script_run '/usr/sbin/uwsgi --plugin python3 --http-socket :7000 --mount /=flask_app:app & sleep 10';
-    assert_script_run 'curl -s http://localhost:7000/ | grep "Hello World"';
-    assert_script_run 'kill $!';
+    # uwsgi is not available on SLE
+    unless (is_sle) {
+        zypper_call "in uwsgi uwsgi-python3";
+        record_info 'uwsgi';
+        assert_script_run '/usr/sbin/uwsgi --plugin python3 --http-socket :7000 --mount /=flask_app:app & sleep 10';
+        assert_script_run 'curl -s http://localhost:7000/ | grep "Hello World"';
+        assert_script_run 'kill $!';
```
Comment (foursixnine) [inline tests/console/flask.pm:53]: This is unrelated change, please move it to a separate commit

File: lib/main_common.pm [hunk 2]
```diff
@@ -1141,22 +1141,8 @@ sub load_console_server_tests {
...
-            loadtest "console/php7_postgresql";
-            loadtest "console/php7_timezone";
-        }
-        else {
-            loadtest "console/php7" unless is_sle;
-            loadtest "console/php8";
-            loadtest "console/php8_mysql";
-            loadtest "console/php8_postgresql";
-            loadtest "console/php8_timezone";
-        }
```
Comment (foursixnine) [inline lib/main_common.pm:1159]: I see you're removing the tests from main_common, but I don't see a shcedule for TW or Leap...

**Review by foursixnine — CHANGES_REQUESTED** [review]

File: lib/main_common.pm (hunk 2 above)
Comment (pdostal) [reply lib/main_common.pm:1159]: You can see it now in the pull request description.

File: tests/console/flask.pm (hunk 1 above)
Comment (pdostal) [reply tests/console/flask.pm:53]: I will do this after I get some other reviews so I don't need to do git magic every time I want to change something in non-last commit.

Comment (foursixnine) [reply tests/console/flask.pm:53]: Cool then :D

File: lib/version_utils.pm [hunk 3]
```diff
@@ -845,3 +846,24 @@ sub get_version_id {
     $cmd = "ssh root\@$args{dst_machine} " . "$cmd" if ($args{dst_machine} ne 'localhost');
     return script_output($cmd);
 }
+
+sub php_version {
+    my ($php, $php_pkg, $php_ver);
+    if (is_leap('<15.0') || is_sle('<15')) {
```
Comment (mimi1vx) [inline lib/version_utils.pm:852]: is really needed `is_leap('<15.0')` ?

**Review by mimi1vx — APPROVED** [review]

File: lib/version_utils.pm (hunk 3 above)
Comment (pdostal) [reply lib/version_utils.pm]: Nope, removed 😄

File: tests/console/flask.pm (hunk 1 above)
Comment (pdostal) [reply tests/console/flask.pm:53]: Done.

**Comment by foursixnine** [comment]
@pdostal can you schedule one for jeos too? I tthink they're running there too (php tests)

**Comment by pdostal** [comment]
@foursixnine we're not scheduling this on JeOS. Can you review please?

**Comment by michaelgrifalconi** [comment]
verification runs did not run the django test. Now django test is failing, working on a fix. https://openqa.suse.de/tests/9621029#step/django/21

**Comment by foursixnine** [comment]
my bad @michaelgrifalconi
